shared/missing.h

```diff
@@ -3,21 +3,17 @@
 #include <unistd.h>
 #include <sys/syscall.h>
 
-#ifdef HAVE_LINUX_MODULE_H
-#include <linux/module.h>
-#endif
-
-#ifndef MODULE_INIT_IGNORE_MODVERSIONS
-# define MODULE_INIT_IGNORE_MODVERSIONS 1
-#endif
-
-#ifndef MODULE_INIT_IGNORE_VERMAGIC
-# define MODULE_INIT_IGNORE_VERMAGIC 2
-#endif
-
-#ifndef MODULE_INIT_COMPRESSED_FILE
-# define MODULE_INIT_COMPRESSED_FILE 4
-#endif
+/*
+ * Macros pulled from linux/module.h, to avoid pulling the header.
+ *
+ * In practise very few people have it installed and distros do not install the
+ * relevant package during their build.
+ *
+ * Values are UAPI, so they cannot change.
+ */
+#define MODULE_INIT_IGNORE_MODVERSIONS 1
+#define MODULE_INIT_IGNORE_VERMAGIC 2
+#define MODULE_INIT_COMPRESSED_FILE 4
```

Comment:
grepping from this of the PR:

```
$ pacman -Ql linux-api-headers | cut -d' ' -f2 | grep -v '.*/$' | sed '/^[[:space:]]*$/d' |  sed 's@/usr/include/@@' | while read f; do git grep  $f; done
shared/missing.h: * Macros pulled from linux/module.h, to avoid pulling the header.
testsuite/module-playground/mod-fake-cciss.c:#include <linux/module.h>
testsuite/module-playground/mod-fake-hpsa.c:#include <linux/module.h>
testsuite/module-playground/mod-fake-scsi-mod.c:#include <linux/module.h>
testsuite/module-playground/mod-foo-a.c:#include <linux/module.h>
testsuite/module-playground/mod-foo-b.c:#include <linux/module.h>
testsuite/module-playground/mod-foo-c.c:#include <linux/module.h>
testsuite/module-playground/mod-foo.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-a.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-b.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-c.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-d.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-e.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-f.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-g.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-h.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-i.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-j.c:#include <linux/module.h>
testsuite/module-playground/mod-loop-k.c:#include <linux/module.h>
testsuite/module-playground/mod-simple.c:#include <linux/module.h>
testsuite/module-playground/mod-weakdep.c:#include <linux/module.h>
```

ok, the remaining ones are the ones from linux-headers. Only caveat is that if we end up including a header and get linux/module.h indirectly include, we break the build. Unlikely though. Maybe just copy the header and adjust the include path to handle it:

shared/missing/linux/module.h

In igt we do a similar thing with the uapi headers: we just pull them in to avoid depending on them (since there we are usually ahead of the any released kernel anyway)